server/catgenome/src/main/java/com/epam/catgenome/util/feature/reader/EhCacheBasedIndexCache.java

```diff
@@ -30,11 +30,13 @@
 import net.sf.ehcache.Ehcache;
 import net.sf.ehcache.Element;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.cache.ehcache.EhCacheCacheManager;
 import org.springframework.stereotype.Service;
 import org.springframework.util.Assert;
 
 @Service
+@ConditionalOnProperty(value = "server.cache.enabled", havingValue = "true")
```

Comment:
I'm not sure if it is sufficient, since a cache is being created anyway in XML config files. Try to find a way to prevent cache creation if it is not needed.